### PR TITLE
Allow to disable training metrics

### PR DESCRIPTION
**Pull request by ohallstrom**
These changes allow to disable the computing of training metrics, and thus only calculate metrics on the validation set (which we most want to consider for the longer training runs).

When `disable_train_metrics` is set to `True` in the config, training is speeded up with close to 3 % with my current training setup (one 8XH100 node with a 130M model).

I also corrected some old formatting with ruff in the files I updated.

### CODE REVIEW

**Comment by warner-benjamin** [comment]
After #87, the only training metric calculation by default should be MaskedAccuracy, that's enough to slow down training by three percent?

**Comment by ohallstrom** [comment]
> After https://github.com/AnswerDotAI/bert24/pull/87, the only training metric calculation by default should be MaskedAccuracy, that's enough to slow down training by three percent?

Sorry, I was not very precise. This was with three metrics  (cross entropy, z loss and masked accuracy), so with only one metric the speed up will not be as big. However, for our main training run, I think we at least want to use two metrics for validation (cross entropy and masked accuracy, so there will still be a noticeable speed up if applied). Else we can ofc make a PR that allows to set the training and eval metrics separately, so that we only have masked accuracy for training metrics, and then cross entropy and masked accuracy for evaluation... 

> I think it's cool to have some metrics during training in addition to evaluation.
We can increase the logging window if that's taking a bit too much time.
We can still merge this as it's only a flag.

I am not sure it is possible to change the frequency of metrics calculation for training without changing the internals of composer. 

Might be subjective, but I don't think training metrics (in addition to loss that always is logged either way) are needed as the info provided from the metrics on validation set should be enough.
